.changeset/forty-crabs-judge.md

```diff
@@ -0,0 +1,30 @@
+---
+'astro': patch
+---
+
+Adds an optional middleware for usage with `astro:env`
+
+If you're using `astro:env`, you can now use a middleware to detect server envrionment variables leaks on the client:
+
+```ts
+import { leakDetectionMiddleware } from 'astro/env/middleware'
+
+export const onRequest = leakDetectionMiddleware()
+```
```

Comment:
nit: I think a better example would to be show how to enable it inside an existing middleware, and tell users that they should put this middleware in the back:
``````suggestion
```ts
// src/middleware.js

import { leakDetectionMiddleware } from 'astro/env/middleware'
import { sequence, defineMiddleware } form 'astro:middleware'

const userMiddleware = defineMiddleware((_, next) => {
	return next()
})

export const onRequest = sequence(
	// have the leak detection utility to run before the user middleware
	userMiddleware,
	leakDetectionMiddleware()
);
```
``````